### PR TITLE
Added new yolo11 head

### LINKED ISSUE #16678: [BUG] `yolov8` pretrained weights are not matching with yaml
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

So, I am training a yolov8n model in coco2017 dataset throughout the last month. I have trained it upto 31 epoch and after that when I tried to train it for the 32th epoch, I saw that at the beginning of training a the message "Transfered 381/453 items from pretrained weight" is shown and amp is checked with YOLO11n model. The model at 31 epoch and before that detects objects in images just fine but the model after 32th epoch shows no detection for the same or different image. I haven't changed anything about the model or dataset configuration as far as I can remember but I am suspicious that maybe yolo v11 realease somehow broke something somewhere. Another thing to note is that, during 32th epoch the losses are comparable to that of the 31th epoch yet the 32th one and even the one's 3 or 4 epoch after that detects nothing. Again from 32 to 34 and onwards I can see "Transfered 453/453 items from pretrained weight". So that makes me suspect the changes somewhere else on the repo as the reason. I'm not sure about it. I would really appreciate it if someone could point out something that I can try.

### Additional

_No response_

### PR DESCRIPTION
**Pull request by ambitious-octopus**
# Test with
## Yolo11
```python
from ultralytics import YOLO

model = YOLO('yolo11n.pt')
ptk = model.model.model.state_dict().keys()

model1 = YOLO('yolo11n.yaml')
yamlk = model1.model.model.state_dict().keys()

for k1, k2 in zip(yamlk, ptk):
    if k1 != k2:
        print(k1, "!=", k2)
```

## Yolo8

```python
from ultralytics import YOLO

model = YOLO('yolov8n.pt')
ptk = model.model.model.state_dict().keys()

model1 = YOLO('yolov8n.yaml')
yamlk = model1.model.model.state_dict().keys()

for k1, k2 in zip(yamlk, ptk):
    if k1 != k2:
        print(k1, "!=", k2)
```
cc: @Y-T-G @Laughing-q 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR introduces a new detection module called `v11Detect` into the Ultralytics YOLO model framework.

### 📊 Key Changes
- Added `v11Detect` to the model's configuration file (`yolo11.yaml`).
- Introduced the `v11Detect` class in the detection head, with modifications to its architecture.
- Updated module imports to include `v11Detect` across relevant files.

### 🎯 Purpose & Impact
- **Enhanced Detection Capabilities**: `v11Detect` aims to improve object detection performance by refining detection head architecture.
- **Increased Flexibility**: Users can now leverage the new detection module for potentially better accuracy and efficiency in their models. 🚀

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
A new detection module, `v11Detect`, has been added to enhance functionality in the Ultralytics YOLO models.

### 📊 Key Changes
- Introduced `v11Detect` class to the codebase.
- Updated configuration files to integrate `v11Detect`.
- Modified detection logic to use `v11Detect`.

### 🎯 Purpose & Impact
- **Advanced Performance:** Improves detection capabilities with more refined convolutional operations.
- **Future-Proofing:** Sets the stage for upcoming enhancements by accommodating newer models.
- **User Impact:** Users will benefit from potentially improved accuracy and performance in object detection tasks. 🚀

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ambitious-octopus, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response, and an Ultralytics engineer will review your submission soon. Meanwhile, to ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `main` branch of the `ultralytics/ultralytics` repository. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Comment by Y-T-G** [comment]
@ambitious-octopus 

We probably need a v11 counterpart for Pose, Segment and OBB heads to inherit from `v11Detect`.

```
In [1]: from ultralytics import YOLO
model
In [2]: model = YOLO('yolo11n-seg.yaml').load('yolo11n-seg.pt')
Transferred 489/525 items from pretrained weights
```

**Comment by ambitious-octopus** [comment]
@Y-T-G You are right, but I am not sure here if it is better to create a yolo11 or a yolov8 head to maintain backwards compatibility. cc: @Laughing-q

**Comment by Laughing-q** [comment]
@ambitious-octopus @Y-T-G Hey guys thanks for the PR and effort here!
I don't know if we want this change or create several v8 heads for backward compatibility since:
- Current version even with updated Detect head, it's still compatible with most use cases(train/val/predict/export modes) of yolov8 pretrained weights or any other custom yolov8 weights trained by users before 8.3.0. For example exporting yolov8 weight with current version it still outputs correct onnx model with v8 head:
  ```bash
  yolo detect export model=yolov8n.pt format=onnx
  ```
- The only case I can think of that would be affected is creating a model from yaml file or/and load the pretrained weight, 
  ```python
  model = YOLO("yolov8n.yaml")
  model = YOLO("yolov8n.yaml").load("yolov8n.pt")
  ```
  which seems like an edge case for me since mostly the training is launched with pretrained weights directly. 
  ```python
  model = YOLO("yolov8n.pt")
  model.train()
  ```
  - The only case that needs creating model from yaml first is users have them customized architecture based on yolov8, but now since yolo11 is out I would encourage them to do architecture modifications/experiments on yolo11 directly.
  - If some users really want to do architecture exploring with v8 models they could also just rollback the version to `8.2.103`.

So the issue here looks like an edge case to me, not sure if we want introduce a tons of code to maintain an edge case that could easily fixed by rollback to `8.2.103`.

**Comment by Ilyabasharov** [comment]
@ambitious-octopus Thanks for your effort! Do you have the metric comparison for the new detection head ?

**Comment by glenn-jocher** [comment]
@Laughing-q yes agree this is complicated as we have many previous models that are deployed now.

**Comment by glenn-jocher** [comment]
The same thing happened before BTW, i.e. when we launched YOLOv8 the v5 and v3 YAMLs inherited the new head, and the models got a lot better, but they were no longer "true" YOLOv3 and YOLOv5 models.

**Comment by glenn-jocher** [comment]
@Ilyabasharov the end-to-end head from YOLOv10 was refined in YOLO11 for improved performance. For detailed metrics, please refer to the YOLO11 documentation.

**Review by ambitious-octopus — COMMENTED** [review]
Closing as not plannet!
